tests/unit/test_sns.py

```diff
@@ -579,3 +583,19 @@ def test_filter_policy_on_message_body(self):
                 assert expected == sub_filter.check_filter_policy_on_message_body(
                     filter_policy, message_body=json.dumps(message_body)
                 ), (filter_policy, message_body)
+
+    @pytest.mark.parametrize("region", ["us-east-1", "eu-central-1", "us-west-2", "my-region"])
+    def test_region_encoded_subscription_token(self, region):
+        token = encode_subscription_token_with_region(region)
+        assert len(token) == 64
+        token_region = get_region_from_subscription_token(token)
+        assert token_region == region
+
+    @pytest.mark.parametrize(
+        "token", ["abcdef123", "mynothexstring", "us-west-2", b"test", b"test2f", "test2f"]
+    )
+    def test_decode_token_with_no_region_encoded(self, token):
+        with pytest.raises(Exception) as e:
```

Comment:
```suggestion
        with pytest.raises(InvalidParameterException) as e:
```
I think being more specific about what exception is raised here might catch possible errors in future.

Reply:
I am matching on the exception under, and if the exception raised does not have the following message `"Invalid parameter: Token"`, the test will fail anyway. Agreed that it could be safer though, I'll add it after the second review, thanks for pointing this out!

Reply:
ah interesting, I didn't see the assert underneath - sorry!

I imagine you already know this (and therefore have a reason for not doing it!) but just in case you can combine that `match` with `pytest.raises` into one line like this:

```
with pytest.raises(InvalidParameterException, match="Invalid parameter: Token"):
    get_region_from_subscription_token(token)
```

Reply:
Thanks for tip! I guess I like having explicit `assert` throughout the test, to quickly spot what we are looking for. But that's a good tip, thanks!